gobblin-core/src/main/java/gobblin/instrumented/converter/InstrumentedConverterBase.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.instrumented.converter;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+import com.google.common.io.Closer;
+
+import gobblin.configuration.WorkUnitState;
+import gobblin.converter.Converter;
+import gobblin.converter.DataConversionException;
+import gobblin.instrumented.Instrumentable;
+import gobblin.instrumented.Instrumented;
+import gobblin.metrics.MetricContext;
+
+
+/**
+ * package-private implementation of instrumentation for {@link gobblin.converter.Converter}.
+ * See {@link gobblin.instrumented.converter.InstrumentedConverter} for extensible class.
+ */
+abstract class InstrumentedConverterBase<SI, SO, DI, DO> extends Converter<SI, SO, DI, DO>
+    implements Instrumentable, Closeable {
+
+  protected MetricContext metricContext = MetricContext.builder("TMP").build();
+  protected Meter recordsIn = new Meter();
+  protected Meter recordsOut = new Meter();
+  protected Meter recordsException = new Meter();
+  protected Timer converterTimer = new Timer();
+  protected Closer closer = Closer.create();
+
+  @Override
+  public Converter<SI, SO, DI, DO> init(WorkUnitState workUnit) {
+    Converter<SI, SO, DI, DO> converter = super.init(workUnit);
+
+    this.metricContext = closer.register(Instrumented.getMetricContext(workUnit, this.getClass()));
+
+    this.recordsIn = this.metricContext.contextAwareMeter("gobblin.converter.records.in");
```

Comment:
Seems you use the type of the metrics, e.g., `Meter` or `Timer`, everywhere else. Please also do that for converter to be consistent.